src/PageContent/FoodLogs/FoodLogsContainer.js

```diff
@@ -1,193 +1,114 @@
+        );
+        const agenciesInfoPromisesList = filteredRawDeliveries.map(rawDelivery => {
+            const agenciesInfoPromise = this.makeAgenciesInfoPromise( 
+                rawDelivery.delivererGroup,
+                rawDelivery.receivingAgency, 
+                rawDelivery.donatingAgency, 
+                rawDelivery.daContact,
+            );
+            return new Promise( async (resolve, reject) => {
+                const agenciesInfo = await Promise.all(agenciesInfoPromise);
+                resolve(agenciesInfo);
+            });
         });
+        const agenciesInfoList = await Promise.all(agenciesInfoPromisesList);
+        this.aggrDeliveries(filteredRawDeliveries, agenciesInfoList);
     }
+
+
+    makeAgenciesInfoPromise(dgId, raId, daId, daContactId) {
+        const dgPromise = new Promise( async (resolve, reject) => {
+            const dgSnapshot = await accountsRef.child(dgId).once('value');
+            resolve(dgSnapshot.val());
+        });
+        const raPromise = new Promise( async (resolve, reject) => {
+            const raSnapshot = await accountsRef.child(raId).once('value');
+            resolve(raSnapshot.val());
+        });
+        const daPromise = new Promise( async (resolve, reject) => {
+            const daSnapshot = await donatingAgenciesRef.child(daId).once('value');
+            resolve(daSnapshot.val());
+        });
+        const daContactPromise = new Promise( async (resolve, reject) => {
+            const daSnapshot = await accountsRef.child(daContactId).once('value');
+            resolve(daSnapshot.val());
+        });
+        return [dgPromise, raPromise, daPromise, daContactPromise]; 
+    }
+    
+    aggrDeliveries(rawDeliveries, agenciesInfoList) {
+        const deliveries = rawDeliveries.map((delivery, i) => {
+            const agenciesInfo = agenciesInfoList[i];
+            const dgInfo = agenciesInfo[0];
+            const raInfo = agenciesInfo[1];
+            const daInfo = agenciesInfo[2];
+            const daContactInfo = agenciesInfo[3];
+            delivery.delivererGroup = dgInfo.name;
+            delivery.receivingAgency = raInfo.name;
+            delivery.donatingAgency = daInfo.name;
+            delivery.daContact = pick(daContactInfo, ['name', 'phone']);
+            return delivery;
+        });
+        this.setState({deliveries: deliveries});
+    }
+
     render(){
         return(
             <div className="food-container ">
                 {/* TODO: Filter feature */}
-                {
+                {this.state.deliveries.length > 0 ?
                     this.state.deliveries.map((completedDelivery, i) => {
                         return (
                             <FoodLogItem delivery={completedDelivery} key={i}/>
                         );
                     })
+                    :
+                    (<h3 className="nothing-found-propmt">No Food Logs Found</h3>)
```

Comment:
It might be nice to be able to differentiate between "No Food Logs Found" and "Actually it's still loading". Maybe set `this.state.deliveries` to be `null` in the constructor? And here we could check whether it's
- `null`: "Loading..."
- `length == 0`: "No Food Logs Found`
- food logs found